src/json_rpc/document_stream.cr

```diff
@@ -24,10 +24,12 @@ module JsonRpc
     # Backing `IO` stream
     getter io : IO
 
+    @buffer_size : Int32 = 0
+
+    private getter buffer : Bytes { Bytes.new(BUFFER_SIZE) }
```

Comment:
Wouldn't a `@buffer = Bytes.new(BUFFER_SIZE)` be easier to grok?

Reply:
`@buffer` does not change its reference, and there was a local assignment [here](https://github.com/Papierkorb/json_rpc/pull/3/files#diff-df89d1ad47d4e78906bb72f2df858fa8e48ce1629468273821530c42d4a80947L43) so I thought a private getter would do the job.

The block to the right is a lazy initialiser.

Reply:
Each `@buffer` will have their own instance with that notation: https://play.crystal-lang.org/#/r/ayzz
Don't ask me why I assigned `@buffer` to a local variable there, probably an oversight after I changed the code before posting it to Github.

Reply:
Getter is a macro for the example you posted https://play.crystal-lang.org/#/r/az3g 

The run here produces equivalent results.

I removed the lazy initialisation as it's almost entirely useless in this situation, as the buffer will almost certainly be used in the lifetime of the class. The lazy initialiser adds a nil check on the underlying `@value`, so introduces unnecessary overhead.